intranet/apps/auth/backends.py

```diff
@@ -65,58 +72,50 @@ def get_kerberos_ticket(username, password):
 
         # We should not try to authenticate with an empty password
         if password == "":
-            return False, False
+            return KerberosAuthenticationResult.FAILURE, False
 
-        cache = "/tmp/ion-%s" % uuid.uuid4()
+        krb5cc_fd, krb5ccname = tempfile.mkstemp(prefix="ion-", text=False)
+        os.close(krb5cc_fd)
 
-        logger.debug("Setting KRB5CCNAME to 'FILE:%s'", cache)
-        os.environ["KRB5CCNAME"] = "FILE:" + cache
+        # Attempt to authenticate against CSL Kerberos realm
+        realm = settings.CSL_REALM
+        result = KerberosAuthenticationBackend.try_single_kinit(
+            username=username, realm=realm, password=password, timeout=settings.KINIT_TIMEOUT, krb5ccname=krb5ccname
+        )
 
         authenticated_through_AD = False
 
-        # Attempt to authenticate against CSL Kerberos realm
+        if result == KerberosAuthenticationResult.FAILURE:
+            # Attempt to authenticate against the Active Directory Kerberos realm
+            authenticated_through_AD = True
+
+            realm = settings.AD_REALM
+            result = KerberosAuthenticationBackend.try_single_kinit(
+                username=username, realm=realm, password=password, timeout=settings.KINIT_TIMEOUT, krb5ccname=krb5ccname
+            )
+
+        if result == KerberosAuthenticationResult.SUCCESS:
+            logger.debug("Kerberos authorized %s@%s - %r", username, realm, authenticated_through_AD)
+        else:
+            logger.debug("Kerberos failed to authorize %s - %r", username, authenticated_through_AD)
+
+        return result, authenticated_through_AD
+
+    @staticmethod
+    def try_single_kinit(*, username: str, realm: str, password: str, krb5ccname: str, timeout: Union[int, float]) -> KerberosAuthenticationResult:
```

Comment:
This shouldn't be addressed in this PR, but the other functions should get type annotations.

Reply:
Yeah, that's more of a thing for a PR addressing #730 broadly.